src/test/java/teammates/test/cases/browsertests/InstructorFeedbackResponseCommentResultsPageUiTest.java

```diff
@@ -0,0 +1,194 @@
+package teammates.test.cases.browsertests;
+
+import org.openqa.selenium.By;
+import org.testng.annotations.Test;
+
+import teammates.common.util.AppUrl;
+import teammates.common.util.Const;
+import teammates.test.pageobjects.InstructorFeedbackResultsPage;
+
+/**
+ * SUT: {@link Const.ActionURIs#INSTRUCTOR_FEEDBACK_RESULTS_PAGE},
+ *      specifically for feedback response comments.
+ */
+public class InstructorFeedbackResponseCommentResultsPageUiTest extends BaseUiTestCase {
+
+    @Override
+    protected void prepareTestData() {
+        testData = loadDataBundle("/InstructorFeedbackResponseCommentResultsPageUiTest.json");
+        removeAndRestoreDataBundle(testData);
+    }
+
+    @Test
+    public void testFeedbackResponseCommentActions() throws Exception {
+
+        InstructorFeedbackResultsPage resultsPage;
+
+        ______TS("Failure case: add empty feedback response comment using comment modal in question view");
+
+        resultsPage = loginToInstructorFeedbackResultsPage("CFResultsUiT.instr", "Open Session");
+        resultsPage.displayByQuestion();
+        resultsPage.loadResultQuestionPanel(1);
+
+        resultsPage.clickCommentModalButton("-2-1-0");
+        resultsPage.addFeedbackResponseCommentInCommentModal("showResponseCommentAddForm-2-1-0", "");
+        resultsPage.verifyCommentFormErrorMessage("-2-1-0", Const.StatusMessages.FEEDBACK_RESPONSE_COMMENT_EMPTY);
```

Comment:
Just curious, how can we achieve this TODO?

Reply:
It means that `addFeedbackResponseCommentInCommentModal` should ensure that the comment has finished adding before returning.

Then we will be able to check for the error message. As for whether there is a clean way to check that the comment has been added correctly that would require further work.

I guess the probably important point here is that asynchronous APIs are harder to reason about than synchronous ones but if they are asynchronous we should document them. 

For example, what if we forgot to wait for the error message and did an immediate check?

In any case, there's no good reason for `addFeedbackResponseCommentInCommentModal` to be asynchronous because any code that follows it usually will check some state related to the comment and then it means almost all variants of the code will have duplicate logic to perform some kind of wait to make sure the comment is added in some way.